_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,298 @@
+  pcr_sequence <- paste0("5; ",
+                         whole_pcr_sequence,
+                         " 3'")
+```
+
+For the next block of code we will begin counting up each nucleotide within the annealing sequence and storing the individual counts of each base. We must account for all plausible bases within the sequence. Not only will we counting the typical A's, T's, C's, and G's, but we need to count up all the ambiguous bases too.
+
+![Ambiguous base chart]({{ site.url }}/assets/pcr-primer-calculator-using-r-ambiguous-chart.jpg)
+[Source](https://www.dnabaser.com/articles/IUPAC%20ambiguity%20codes.html)
+
+```r
+  # get a count of each nucleotide within the sequence
+  g_cnt <- stringr::str_count(annealing_seq, "G")
+  c_cnt <- stringr::str_count(annealing_seq, "C")
+  t_cnt <- stringr::str_count(annealing_seq, "T")
+  a_cnt <- stringr::str_count(annealing_seq, "A")
+
+  # get a count of each ambiguous base within the sequence
+  y_cnt <- stringr::str_count(annealing_seq, "Y")
+  r_cnt <- stringr::str_count(annealing_seq, "R")
+  w_cnt <- stringr::str_count(annealing_seq, "W")
+  s_cnt <- stringr::str_count(annealing_seq, "S")
+  k_cnt <- stringr::str_count(annealing_seq, "K")
+  m_cnt <- stringr::str_count(annealing_seq, "M")
+  d_cnt <- stringr::str_count(annealing_seq, "D")
+  v_cnt <- stringr::str_count(annealing_seq, "V")
+  h_cnt <- stringr::str_count(annealing_seq, "H")
+  b_cnt <- stringr::str_count(annealing_seq, "B")
+  n_cnt <- stringr::str_count(annealing_seq, "N")
+
+```
+
+Now that we have a count of each possible base for a given annealing primer sequence, we can calculating the melting temperature of each primer. Below is the equation for finding the melting temperature of a given sequence. As you can see, the melting point is decided by the amount of A-T bases and G-C bases. But how does ambiguous bases fit into the equation? This is a tricky question to answer because ambiguous bases can become another bases during PCR (refer to the *Ambiguous base chart* for each ambiguous base to see which nucleotide that base could become). We will need to calculate two melting temperatures; a minimum and maximum temperature for sequences with ambiguous bases. To calculate the minimum temperature, we will count the ambiguous bases that could potentially become an A or T base as an A or T nucleotide. The ambiguous bases that can **only** become a G or C will be counted as a G or C nucleotide. For calculating the maximum temperature, we will flip the ambiguous assumption, in which ambiguous bases that could potential become a G or C base will be counted as a G or C nucleotide. And ambiguous bases that can **only** become an A or T base will be counted as an A or T nucleotide. To make this concept easier to conceptualize, I have include the pseudocode before the code block.
+
+> Melting Temp = 2 * (A + T) + 4 * (G + C)
+
+
+**Pseudocode for calculating melting temperature**  
```

Comment:
It seems like some of your text is reading like code formatting. Try viewing this as html to catch this.

Reply:
@naupaka would you know why the "for" is in red? This happens two more times in my .md file. (on line 147)